draft-ietf-quic-load-balancers.md

```diff
@@ -1198,6 +1198,87 @@ format and cryptographic protection of this information may change in future
 versions or extensions of TLS or QUIC, and therefore this functionality is
 inherently not version-invariant.
 
+# Interoperability with DTLS over UDP
+
+Some environments may contain DTLS traffic as well as QUIC operating over UDP,
+which may be hard to distinguish.
+
+In most cases, the packet parsing rules above will cause a QUIC-LB load
+balancer to route DTLS traffic in an appropriate way. DTLS 1.3 implementations
+that use the connection_id extension [?I-D.ietf-tls-dtls-connection-id] might
+use the techniques in this document to generate connection IDs and achieve
+robust routability for DTLS associations if they meet a few additional
+requirements.
+
+## DTLS 1.0 and 1.2
+
+DTLS 1.0 [?RFC4347] and 1.2 [?RFC6347] use packet formats that a QUIC-LB router
+will interpret as short header packets with CIDs that request 4-tuple routing.
+As such, they will route such packets consistently as long as the 4-tuple does
+not change. Note that DTLS 1.0 has been deprecated by the IETF.
+
+The first octet of every DTLS 1.0 or 1.2 datagram contains the content type.
+A QUIC-LB load balancer will interpret any content type less than 128 as a short
+header packet, meaning that the subsequent octets should contain a connection
+ID.
+
+Existing TLS content types comfortably fit in the range below 128. Assignment of
+codepoints greater than 64 would require coordination in accordance with
+[?RFC7983], and anyway would likely create problems demultiplexing DTLS and
+version 1 of QUIC. Therefore, this document believes it is extremely unlikely
+that TLS content types of 128 or greater will be assigned. Nevertheless, such
+an assignment would cause a QUIC-LB load balancer to interpret the packet as a
+QUIC long header with an essentially random connection ID, which is likely to be
+routed irregularly.
+
+The second octet of every DTLS 1.0 or 1.2 datagram is the bitwise complement
+of the DTLS Major version (i.e. version 1.x = 0xfe). A QUIC-LB load balancer
+will interpret this as a connection ID that requires 4-tuple based load
+balancing, meaning that the routing will be consistent as long as the 4-tuple
+remains the same.
+
+[?I-D.ietf-tls-dtls-connection-id] defines an extension to add connection IDs
+to DTLS 1.2. Unfortunately, a QUIC-LB load balancer will not correctly parse
+the connection ID and will continue 4-tuple routing. An modified QUIC-LB load
+balancer that correctly identifies DTLS and parses a DTLS 1.2 datagram for
+the connection ID is outside the scope of this document.
```

Comment:
It seems like, given the amount of text you just added for DTLS it would be trivial to identify how to match the first DTLS octet and then find the connection ID.  It's just a few bytes down at a fixed location.

Reply:
I guess, but then you have to talk about demultiplexing QUIC and DTLS, the bit-grease extension, and you're breaking the almost-version-invariance of QUIC-LB pretty badly.

This draft has already got too much stuff in it, so I thought I'd just write down how DTLS would work out of the box. If people care about this use case I could write DTLS-LB or whatever.